beps/0003-auth-architecture-evolution/README.md

```diff
@@ -97,25 +99,105 @@ export interface AuthService {
 }
 ```
 
-#### Usage Patterns
+### `AuthService` Usage Patterns
 
 TODO
 
-### `HttpAuthService` (WIP)
+### `HttpRouterService` Interface
 
-The new `HttpAuthService` interface is defined as follows:
+> Open question: Should this instead be added to the `HttpAuthService`? It may fit a bit better there, but on the other hand it might make sense to add additional policies unrelated to authentication too, such as rate limiting.
 
-```ts
-type AuthTypes = 'user' | 'user-cookie' | 'service' | 'unauthorized';
+The `HttpRouterService` interface will be extended with the ability to opt-out of the default protection of endpoints, enabling either cookie auth or unauthenticated access.
 
-export interface HttpAuthServiceMiddlewareOptions {
-  allow: AuthTypes[];
+```ts
+export interface HttpRouterService {
+  // All routes only allow authenticated users and services by default.
+  use(handler: Handler): void;
```

Comment:
One of the things I was thinking about in my PR, does the current pattern of exporting a single giant router mesh well with default authentication? Would it make life easier to instead encourage a patter where `createRouter: (options)=>express.Router` becomes `createRouters: (options) => {router: express.Router, unauthenticatedRouter: express.Router, cookieAuthRouter: express.Router}` or something similar.

Reply:
The `createRouter` pattern is overall something that should disappear imo, at least as something semi-required. It ofc needs to be there still in order to support the old system, so we can't really make that exact change.

Thinking about this a bit more though I think we might be able to get away with having a couple of different routers created internally for the few plugin that need it. Then we just wire all of that up into a single router in `createRouter` for the old system, but we don't use that. Initially we were thinking that that's gonna get quite messy, but given the assumption that most plugins only need default auth (user + service), that should be manageable.

If you want to you could open a PR to tweak the BEP to use separate routers and we could discuss it there with some more context. I think it might work.

We had some ideas around applying additional policies to individual routes, but tbh that might as well be some kind of `HttpPolicyService` or w/e, and not have that cover authentication, so nvm, that part